morph_tool/morphdb.py

```diff
@@ -0,0 +1,371 @@
+        self.layer = str(layer or '')
+
+        for attr in BOOLEAN_REPAIR_ATTRS:
+            setattr(self, attr, kwargs.get(attr, True))
+
+        self.axon_inputs = kwargs.get('axon_inputs', [])
+        self.path = kwargs.get('path')
+        self.label = kwargs.get('label')
+
+    @classmethod
+    def _from_xmldict(cls, item: Dict[str, Any]):
+        '''MorphInfo ctor.
+        Args:
+            item: A dictionary that represents the content of the XML file.
+                  The only mandatory keys are [name, mtype, layer]
+        '''
+        morph = cls(
+            item['name'],
+            cls.MTYPE_SEPARATOR.join(filter(None, [item['mtype'], item.get('msubtype')])),
+            item['layer']
+        )
+
+        def is_true(repair: Dict[str, Any], key: str) -> bool:
+            '''Parse a string representing a boolean repair flag and returns its boolean value
+            Unless clearly stated as false, missing tags default to True
+            - According to Eilif, an empty use_axon (corresponding to a null in the database)
+              means that the axon is supposed to be used
+            - dendrite_repair       defaults to True in BlueRepairSDK
+            - basal_dendrite_repair defaults to True in BlueRepairSDK
+            - unravel: well I guess we always want to do it
+            '''
+            el = repair.get(key)
+            if el not in {'true', 'false', 'True', 'False', None}:
+                raise ValueError(f'Invalid XML element {key} has invalid value: {el}\n'
+                                 'Allowed values:\n'
+                                 '- empty tag (which is equivalent to True)\n',
+                                 '- true\n',
+                                 '- True\n',
+                                 '- false\n',
+                                 '- False')
+            return el in (None, 'true', 'True', )
+
+        repair = item.get('repair', {})
+        for attr in BOOLEAN_REPAIR_ATTRS:
+            setattr(morph, attr, is_true(repair, attr))
+
+        # "always_iterable" deals with <axoninput> not being interpreted
+        # as a list if there is a single entry <axoninput> entry in the XML.
+        morph.axon_inputs = list(always_iterable(
+            (repair.get('axon_sources') or {}).get('axoninput', [])
+        ))
+
+        return morph
+
+    @property
+    def row(self) -> List:
+        '''Flattened data structude ready to be used by a dataframe.'''
+        return [getattr(self, attr) for attr in COLUMNS]
+
+    def __repr__(self):
+        return (f'MorphInfo(name={self.name}, mtype={self.mtype}, layer={self.layer}, '
+                f'label={self.label})')
+
+
+class MorphDB:
+    '''A MorphInfo container.
+    It takes care of maintaining uniqueness of the MorphInfo element
+    and methods to write neurondb to various format (xml, dat, csv)
+    '''
+
+    def __init__(self, morph_info_seq: Optional[Sequence[MorphInfo]] = None):
```

Comment:
I'm answering here to https://github.com/BlueBrain/morph-tool/pull/25#discussion_r565978296 because it's in outdated code.
Yes I think `db += MorphDB(iterable)` is good. The `update` method could be useful if some optimization is possible, but it seems to me that we need to concatenate the dataframes and create a new dataframe in either way.